### PR TITLE
Feature/conan logger

### PR DESCRIPTION
**Pull request by lasote**
- Create a log-trace with conan actions. Logged if ``CONAN_TRACE_FILE`` points to a file.
- Improved runner to optionally log to file the "self.run" commands, controlled with those variables:
    - CONAN_PRINT_RUN_COMMANDS: Print the commands to output e.j:``*****Running*******\n > cmake``
    - CONAN_LOG_RUN_TO_FILE: Create a log file in the current folder (source, build, etc) appending each self.run execution. Will use PIPE
    - CONAN_LOG_RUN_TO_OUTPUT:  Default True. If False won't log any output. (Will use PIPE and discard the output)
- If CONAN_LOG_RUN_TO_OUTPUT and not CONAN_LOG_RUN_TO_FILE and not used special StrinIO in self.run command, sys.command will be used (default)

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -735,6 +747,7 @@ def user(self, *parameters):
         parser.add_argument('-c', '--clean', default=False,
                             action='store_true', help='Remove user and tokens for all remotes')
         args = parser.parse_args(*parameters)  # To enable -h
+        log_command("user", vars(args))
```
Comment (memsharded) [inline conans/client/command.py]: Please intercept and remove the password field from logging

File: conans/client/rest/rest_client.py [hunk 2]
```diff
@@ -367,19 +374,24 @@ def _post_json(self, url, payload):
...
                                            data=json.dumps(data))
         else:
-            response = self.requester.get(url, auth=self.auth, headers=self.custom_headers,
+            response = self.requester.get(url, auth=self.auth, headers=headers,
                                           verify=self.verify_ssl,
                                           stream=True)
+
+        duration = time.time() - t1
+        method = "POST" if data else "GET"
+        log_client_rest_api_call(url, method, duration, headers)
```
Comment (memsharded) [inline conans/client/rest/rest_client.py]: If these headers will carry basic-auth to get JWT, please remove that header from logging

File: conans/model/conan_file.py [hunk 3]
```diff
@@ -113,6 +115,7 @@ def __init__(self, output, runner, settings, conanfile_directory, user=None, cha
 
         self._conanfile_directory = conanfile_directory
         self.package_folder = None  # Assigned at runtime
+        self.log_folder = None  # Assigned at runtime
```
Comment (memsharded) [inline conans/model/conan_file.py]: Is this variable used somewhere?

Comment (lasote) [inline conans/model/conan_file.py]: Nope. Removing

**Pull request merged**